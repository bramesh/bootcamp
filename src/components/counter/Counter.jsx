import React from "react";
import PropTypes from 'prop-types';

class Counter extends React.PureComponent {
    static propTypes = {
        count: PropTypes.number
    }

    static defaultProps = {
        count: 0
    }
    
    render() {
        const {
            count,
            onCountChange,
            reset,
            ...elementProps
        } = this.props;
        return(
            <div {...elementProps} className="counter">  
                {!count && (<h2>Click the button to increment the count</h2>)}
                <h1>{count}</h1>
                <button className="incrementButton" onClick={onCountChange}>Increment</button>
                <button className="resetButton" onClick={reset}>Reset counter</button>
            </div>
        )
    }
}

export default Counter;