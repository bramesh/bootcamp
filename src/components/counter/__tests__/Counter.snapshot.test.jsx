import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../Counter';

describe('Counter snapshot tests', () => {
  it('render counter with default count', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('render counter with 5', () => {
    const tree = renderer.create(<Counter count={5} />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})