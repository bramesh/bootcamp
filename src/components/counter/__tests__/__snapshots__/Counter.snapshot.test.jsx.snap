// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot tests render counter with 5 1`] = `
<div
  className="counter"
>
  <h1>
    5
  </h1>
  <button
    className="incrementButton"
  >
    Increment
  </button>
  <button
    className="resetButton"
  >
    Reset counter
  </button>
</div>
`;

exports[`Counter snapshot tests render counter with default count 1`] = `
<div
  className="counter"
>
  <h2>
    Click the button to increment the count
  </h2>
  <h1>
    0
  </h1>
  <button
    className="incrementButton"
  >
    Increment
  </button>
  <button
    className="resetButton"
  >
    Reset counter
  </button>
</div>
`;
